lib/matplotlib/_api/deprecation.pyi

```diff
@@ -0,0 +1,79 @@
+from collections.abc import Callable
+import contextlib
+from typing import Any, Literal, TypedDict, TypeVar, overload
+from typing_extensions import (
+    ParamSpec,  # < Py 3.10
+    Unpack,  # < Py 3.11
+)
+
+_P = ParamSpec("_P")
+_R = TypeVar("_R")
+_T = TypeVar("_T")
+
+class MatplotlibDeprecationWarning(DeprecationWarning): ...
+
+class DeprecationKwargs(TypedDict, total=False):
+    message: str
+    alternative: str
+    pending: bool
+    obj_type: str
+    addendum: str
+    removal: str
+
+class NamedDeprecationKwargs(DeprecationKwargs, total=False):
+    name: str
+
+def warn_deprecated(since: str, **kwargs: Unpack[NamedDeprecationKwargs]) -> None: ...
+def deprecated(
+    since: str, **kwargs: Unpack[NamedDeprecationKwargs]
+) -> Callable[[_T], _T]: ...
+
+class deprecate_privatize_attribute(Any):
+    def __init__(self, since: str, **kwargs: Unpack[NamedDeprecationKwargs]): ...
+    def __set_name__(self, owner: type[object], name: str) -> None: ...
+
+DECORATORS: dict[Callable, Callable] = ...
+
+@overload
+def rename_parameter(
+    since: str, old: str, new: str, func: Literal[None] = None
```

Comment:
```suggestion
    since: str, old: str, new: str, func: None=...
```

None, being a singleton, has no need to be called out as `Literal[None]`.
Additionally, pyi files call for the actual value of defaults to be omitted (which admittedly makes little difference when the type hint is `None`...)

Same applies to all other uses of `Literal` in this file.

Reply:
It's not _wrong_ just a bit out of place/using tools that are not needed for the task at hand